events/management/commands/add_yso_audience.py

```diff
@@ -0,0 +1,82 @@
+from functools import lru_cache
+
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+
+from events.models import Event, Keyword, KeywordSet
+
+YSO_KEYWORD_SET_DATA = {
+    'id': 'audiences',
+    'name_en': 'audiences',
+    'name_fi': 'kohderyhmät',
+    'name_sv': 'invånargrupper',
+    'data_source_id': 'yso',
+    'usage': KeywordSet.AUDIENCE,
+}
+
+# keyword id mapping from hel.fi to YSO
+KEYWORD_MAPPING = {
+    'helfi:1': ['yso:p4354', 'yso:p13050'],  # lapset ja lapsiperheet -> lapset (ikään liittyvä rooli) & lapsiperheet
+    'helfi:2': ['yso:p11617'],  # nuoret -> nuoret (ikään liittyvä rooli)
+    'helfi:3': ['yso:p6165'],  # maahanmuuttajat -> maahanmuuttajat
+    'helfi:4': ['yso:p7179'],  # vammaiset -> vammaiset
+    'helfi:5': ['yso:p2434'],  # vanhukset -> vanhukset
+    'helfi:6': ['yso:p3128'],  # yritykset -> yritykset
+    'helfi:7': ['yso:p1393'],  # yhdistykset -> järjestöt
+}
+
+
+class Command(BaseCommand):
+    help = "Creates YSO audience keyword set and adds YSO audience keywords to events."
+
+    @lru_cache()
+    def get_keyword_obj(self, keyword_id):
+        try:
+            keyword = Keyword.objects.get(id=keyword_id)
+        except Keyword.DoesNotExist:
+            raise CommandError('keyword "%s" does not exist' % keyword_id)
+        return keyword
+
+    @transaction.atomic()
+    def create_yso_audiences_keyword_set(self):
+        self.stdout.write('creating YSO audiences keyword set...')
+
+        # create the set itself
+        yso_keyword_set, created = KeywordSet.objects.update_or_create(
+            id=YSO_KEYWORD_SET_DATA['id'],
+            defaults=YSO_KEYWORD_SET_DATA
+        )
+        if created:
+            self.stdout.write('created keyword set "%s"' % YSO_KEYWORD_SET_DATA['id'])
+        else:
+            self.stdout.write('keyword set "%s" already exist' % YSO_KEYWORD_SET_DATA['id'])
+
+        # add keywords to the set
+        for key, yso_keyword_ids in KEYWORD_MAPPING.items():
+
+            for yso_keyword_id in yso_keyword_ids:
+                yso_keyword = self.get_keyword_obj(yso_keyword_id)
+
+                if yso_keyword not in yso_keyword_set.keywords.all():
```

Comment:
Does Django cache the calls? I think this might end up in many hits to the database. You could just save the kw set before the for loop:

```python
yso_kw_set = set(yso_keyword_set.keywords.all())
```